File: python/0238.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        output = [0 for i in range(len(nums))]
        output[0] = nums[0]
        for i in range(1,len(nums)):
            output[i] = output[i-1] * nums[i]
        right_multiplier = 1
        for i in range(len(nums)-1, 0, -1):
            output[i] = output[i-1] * right_multiplier
            right_multiplier *= nums[i]
        output[0] = right_multiplier
        return output

File: go/0238.go
package leetcode

func productExceptSelf(nums []int) []int {
	products := make([]int, len(nums))

	temp_products := 1
	for idx := 0; idx < len(nums); idx++ {
		products[idx] = temp_products
		temp_products *= nums[idx]
	}

	temp_products = 1
	for idx := len(products) - 2; idx >= 0; idx-- {
		temp_products *= nums[idx+1]
		products[idx] *= temp_products
	}

	return products
}
